Plugins/CoreAPI/server/graphql/resolvers/mutations.js

```diff
@@ -0,0 +1,435 @@
+import PenPal from "meteor/penpal";
+import { Mongo } from "meteor/mongo";
+import _ from "lodash";
+
+export default {
+  async setCoreAPIConfiguration(root, args, context) {
+    let currConfig = PenPal.DataStore.fetch("CoreAPI", "Configuration", {});
+    if (currConfig.length > 0) {
+      PenPal.DataStore.update(
+        "CoreAPI",
+        "Configuration",
+        { _id: `${currConfig[0]._id}` },
+        { $set: { hookURL: args.hookURL } }
+      );
+      return {
+        status: "Updated Configuration",
+        was_success: true,
+        affected_records: [currConfig[0]._id],
+      };
+    } else {
+      let addedConfig = PenPal.DataStore.insert("CoreAPI", "Configuration", {
+        hookURL: args.hookURL,
+      });
+      if (addedConfig) {
+        return {
+          status: "Inserted Configuration",
+          was_success: true,
+          affected_records: [addedConfig],
+        };
+      } else {
+        return {
+          status: "Configuration Update Failed",
+          was_success: false,
+          affected_records: [],
+        };
+      }
+    }
+  },
+  async upsertHosts(root, args, context) {
+    // Check if we already have the host
```

Comment:
Are we also going to abstract these into a `PenPal.API.Host.Upsert` or something like that at some point? GraphQL resolver having this code doesn't make it particularly accessible to other plugins as an API.

Reply:
yes